core/templates/pages/skill-editor-page/skill-editor-page.component.ts

```diff
@@ -83,8 +84,24 @@ angular.module('oppia').component('skillEditorPage', {
       ctrl.getWarningsCount = function() {
         return ctrl.skill.getValidationIssues().length;
       };
+
+      ctrl.setUpBeforeUnload = function() {
+        WindowRef.nativeWindow.addEventListener(
+          'beforeunload', ctrl.confirmBeforeLeaving);
+      };
+
+      ctrl.confirmBeforeLeaving = function(e) {
+        if (UndoRedoService.getChangeCount()) {
+          // This message is irrelevant, but is needed to trigger the
+          // confirmation before leaving.
+          e.returnValue = 'Sure?';
+          return false;
+        }
+      };
```

Comment:
What browsers have you tested this change in?

Reply:
Chrome and firefox.

Reply:
cool, thanks!